src/ert/scheduler/scheduler.py

```diff
@@ -228,9 +231,10 @@ async def _monitor_and_handle_tasks(
                         continue
                     if task_exception := task.exception():
                         raise task_exception
-                # make sure to empty report all events before exit
-                while not self._events.empty():
-                    await asyncio.sleep(0)
+                # make sure to report all events before exit
+                if self._publisher_task_active:
```

Comment:
Should we again set `self._publisher_task_active = False` after this if block is executed?

Reply:
I was looking at a specific commit rather than the entire change...

Reply:
Now it's gone - I've reverted it back as described in the comment https://github.com/equinor/ert/pull/7562#discussion_r1553047562